packages/rocketchat-custom-oauth/server/custom_oauth_server.js

```diff
@@ -180,64 +240,9 @@ export class CustomOAuth {
 					identity = identity.result;
 				}
 
-				// Fix WordPress-like identities having 'ID' instead of 'id'
-				if (identity.ID && !identity.id) {
-					identity.id = identity.ID;
-				}
-
-				// Fix Auth0-like identities having 'user_id' instead of 'id'
-				if (identity.user_id && !identity.id) {
-					identity.id = identity.user_id;
-				}
-
-				if (identity.CharacterID && !identity.id) {
-					identity.id = identity.CharacterID;
-				}
-
-				// Fix Dataporten having 'user.userid' instead of 'id'
-				if (identity.user && identity.user.userid && !identity.id) {
-					if (identity.user.userid_sec && identity.user.userid_sec[0]) {
-						identity.id = identity.user.userid_sec[0];
-					} else {
-						identity.id = identity.user.userid;
-					}
-					identity.email = identity.user.email;
-				}
-				// Fix for Xenforo [BD]API plugin for 'user.user_id; instead of 'id'
-				if (identity.user && identity.user.user_id && !identity.id) {
-					identity.id = identity.user.user_id;
-					identity.email = identity.user.user_email;
-				}
-				// Fix general 'phid' instead of 'id' from phabricator
-				if (identity.phid && !identity.id) {
-					identity.id = identity.phid;
-				}
-
-				// Fix Keycloak-like identities having 'sub' instead of 'id'
-				if (identity.sub && !identity.id) {
-					identity.id = identity.sub;
-				}
-
-				// Fix general 'userid' instead of 'id' from provider
-				if (identity.userid && !identity.id) {
-					identity.id = identity.userid;
-				}
-
-				// Fix Nextcloud provider
-				if (!identity.id && identity.ocs && identity.ocs.data && identity.ocs.data.id) {
-					identity.id = identity.ocs.data.id;
-					identity.name = identity.ocs.data.displayname;
-					identity.email = identity.ocs.data.email;
-				}
-
-				// Fix when authenticating from a meteor app with 'emails' field
-				if (!identity.email && (identity.emails && Array.isArray(identity.emails) && identity.emails.length >= 1)) {
-					identity.email = identity.emails[0].address ? identity.emails[0].address : undefined;
-				}
+				self.fixThirdPartyIdentityRules(identity);
```

Comment:
should this return the new value instead of modifying the parameter?

Reply:
I changed this code further on #11856, it's easier to just close this PR and wait for the wechat integration to be merged.

Reply:
alright, closing then =) thx